mutt/charset.c

```diff
@@ -646,7 +646,10 @@ iconv_t mutt_ch_iconv_open(const char *tocode, const char *fromcode, uint8_t fla
     /* get rid of the oldest entry */
     FREE(&IconvCache[IconvCacheUsed - 1].fromcode1);
     FREE(&IconvCache[IconvCacheUsed - 1].tocode1);
-    iconv_close(IconvCache[IconvCacheUsed - 1].cd);
+    if (iconv_t_valid(IconvCache[IconvCacheUsed - 1].cd))
+    {
+      iconv_close(IconvCache[IconvCacheUsed - 1].cd);
+    }
```

Comment:
We probably need this in `mutt_ch_cache_cleanup()` too.
https://github.com/neomutt/neomutt/blob/a8cd45ed557013f6166d485bc10bf31d021c4c28/mutt/charset.c#L1166-L1175